File: BinaryTree/csharp/TreeMin.cs
using System;

namespace BinaryTreeSolutions;

public static class TreeMin
{
    public static int Solve(TreeNode? root)
    {
        if (root is null)
        {
            return int.MaxValue;
        }

        return Math.Min(root.Val, Math.Min(Solve(root.Left), Solve(root.Right)));
    }
}

File: BinaryTree/java/treeMin.java
class TreeMin {
  public static int treeMin(TreeNode root) {
    if (root == null) {
      return Integer.MAX_VALUE;
    }
    return Math.min(root.val, Math.min(treeMin(root.left), treeMin(root.right)));
  }

  public static void run() {
    // intentionally empty
  }
}
